### PR TITLE
draft FIPS enabled Ubuntu 20.04 Distros

### PR DESCRIPTION
**Pull request by calvin197**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Comment by calvin197** [comment]
Linus VHD Build test
https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=69525056&view=results

**Comment by calvin197** [comment]
Work items: 
https://msazure.visualstudio.com/CloudNativeCompute/_workitems/edit/17405512/
https://msazure.visualstudio.com/CloudNativeCompute/_workitems/edit/17405525/

File: parts/linux/cloud-init/artifacts/sshd_config_2004_fips [hunk 1]
```diff
@@ -0,0 +1,125 @@
+#	$OpenBSD: sshd_config,v 1.101 2017/03/14 07:19:07 djm Exp $
```
Comment (calvin197) [inline parts/linux/cloud-init/artifacts/sshd_config_2004_fips:1]: It's copied from "parts/linux/cloud-init/artifacts/sshd_config_1804_fips". Not sure if it's the correct config.

File: .pipelines/.vsts-vhd-builder-release.yaml [hunk 2]
```diff
@@ -86,6 +86,14 @@ parameters:
...
   type: boolean
   default: true
+- name: build2004fipscontainerd
+  displayName: Build 2004 FIPS containerd
+  type: boolean
+  default: true
+- name: build2004fipsgen2containerd
+  displayName: Build 2004 FIPS Gen2 containerd
+  type: boolean
+  default: true
```
Comment (chengliangli0918) [inline .pipelines/.vsts-vhd-builder-release.yaml:96]: let's use false here before we finalize everything and want to include this vhd build in weekly aks vhd build.

File: .pipelines/.vsts-vhd-builder-release.yaml [hunk 3]
```diff
@@ -86,6 +86,14 @@ parameters:
   displayName: Build 1804 FIPS Gen2 GPU+containerd
   type: boolean
   default: true
+- name: build2004fipscontainerd
+  displayName: Build 2004 FIPS containerd
+  type: boolean
+  default: true
```
Comment (chengliangli0918) [inline .pipelines/.vsts-vhd-builder-release.yaml:92]: let's use false here before we finalize everything and want to include this vhd build in weekly aks vhd build.

File: .pipelines/.vsts-vhd-builder-release.yaml [hunk 4]
```diff
@@ -637,6 +645,57 @@ stages:
...
+    dependsOn: []
+    condition: eq('${{ parameters.build2004fipscontainerd }}', true)
+    jobs:
+    - job: build
+      timeoutInMinutes: 180
+      steps:
+        - bash: |
+            echo '##vso[task.setvariable variable=DRY_RUN]${{parameters.dryrun}}'
+            echo '##vso[task.setvariable variable=OS_SKU]Ubuntu'
+            echo '##vso[task.setvariable variable=OS_VERSION]18.04'
```
Comment (chengliangli0918) [inline .pipelines/.vsts-vhd-builder-release.yaml:658]: 20.04

File: .pipelines/.vsts-vhd-builder-release.yaml [hunk 5]
```diff
@@ -637,6 +645,57 @@ stages:
...
+            echo '##vso[task.setvariable variable=IMG_PUBLISHER]Canonical'
+            echo '##vso[task.setvariable variable=IMG_OFFER]UbuntuServer'
```
Comment (chengliangli0918) [inline .pipelines/.vsts-vhd-builder-release.yaml:660]: UbuntuServer

File: .pipelines/.vsts-vhd-builder-release.yaml [hunk 6]
```diff
@@ -637,6 +645,57 @@ stages:
...
```
Comment (chengliangli0918) [inline .pipelines/.vsts-vhd-builder-release.yaml:685]: 0001-com-ubuntu-server-focal

File: .pipelines/.vsts-vhd-builder-release.yaml [hunk 7]
```diff
@@ -637,6 +645,57 @@ stages:
...
+    dependsOn: []
+    condition: eq('${{ parameters.build2004fipsgen2containerd }}', true)
+    jobs:
+    - job: build
+      timeoutInMinutes: 180
+      steps:
+        - bash: |
+            echo '##vso[task.setvariable variable=DRY_RUN]${{parameters.dryrun}}'
+            echo '##vso[task.setvariable variable=OS_SKU]Ubuntu'
+            echo '##vso[task.setvariable variable=OS_VERSION]18.04'
```
Comment (chengliangli0918) [inline .pipelines/.vsts-vhd-builder-release.yaml:683]: 20.04

File: .pipelines/.vsts-vhd-builder-release.yaml [hunk 8]
```diff
@@ -637,6 +645,57 @@ stages:
...
+            echo '##vso[task.setvariable variable=IMG_PUBLISHER]Canonical'
+            echo '##vso[task.setvariable variable=IMG_OFFER]UbuntuServer'
+            echo '##vso[task.setvariable variable=IMG_SKU]18.04-LTS'
```
Comment (chengliangli0918) [inline .pipelines/.vsts-vhd-builder-release.yaml:661]: 20_04-lts

File: .pipelines/.vsts-vhd-builder-release.yaml [hunk 9]
```diff
@@ -637,6 +645,57 @@ stages:
...
+    - job: build
+      timeoutInMinutes: 180
+      steps:
+        - bash: |
+            echo '##vso[task.setvariable variable=DRY_RUN]${{parameters.dryrun}}'
+            echo '##vso[task.setvariable variable=OS_SKU]Ubuntu'
+            echo '##vso[task.setvariable variable=OS_VERSION]18.04'
+            echo '##vso[task.setvariable variable=IMG_PUBLISHER]Canonical'
+            echo '##vso[task.setvariable variable=IMG_OFFER]UbuntuServer'
+            echo '##vso[task.setvariable variable=IMG_SKU]18_04-LTS-GEN2'
```
Comment (chengliangli0918) [inline .pipelines/.vsts-vhd-builder-release.yaml:686]: 20_04-lts-gen2

File: parts/linux/cloud-init/artifacts/sshd_config_2004_fips (hunk 1 above)
Comment (chengliangli0918) [reply parts/linux/cloud-init/artifacts/sshd_config_2004_fips:1]: we don't need special sshd_config file on 20.04 fips as 18.04 fips requires. So this file can be removed.

File: vhdbuilder/packer/packer_source.sh [hunk 10]
```diff
@@ -190,6 +190,8 @@ copyPackerFiles() {
     SSHD_CONFIG_SRC=/home/packer/sshd_config_1604
   elif [[ ${UBUNTU_RELEASE} == "18.04" && ${ENABLE_FIPS,,} == "true" ]]; then
     SSHD_CONFIG_SRC=/home/packer/sshd_config_1804_fips
+  elif [[ ${UBUNTU_RELEASE} == "20.04" && ${ENABLE_FIPS,,} == "true" ]]; then
```
Comment (chengliangli0918) [inline vhdbuilder/packer/packer_source.sh:193]: no special sshd_config is required for 20.04 fips.

File: vhdbuilder/packer/vhd-image-builder-arm64-gen2.json [hunk 11]
```diff
@@ -290,6 +289,11 @@
       "source": "parts/linux/cloud-init/artifacts/sshd_config_1804_fips",
       "destination": "/home/packer/sshd_config_1804_fips"
     },
+    {
```
Comment (chengliangli0918) [inline vhdbuilder/packer/vhd-image-builder-arm64-gen2.json:292]: can be removed.

File: vhdbuilder/packer/vhd-image-builder-base.json [hunk 12]
```diff
@@ -301,6 +301,11 @@
       "source": "parts/linux/cloud-init/artifacts/sshd_config_1804_fips",
       "destination": "/home/packer/sshd_config_1804_fips"
     },
+    {
```
Comment (chengliangli0918) [inline vhdbuilder/packer/vhd-image-builder-base.json:304]: can be removed

File: vhdbuilder/packer/vhd-image-builder-sigminimal.json [hunk 13]
```diff
@@ -294,6 +294,11 @@
       "source": "parts/linux/cloud-init/artifacts/sshd_config_1804_fips",
       "destination": "/home/packer/sshd_config_1804_fips"
     },
+    {
+      "type": "file",
```
Comment (chengliangli0918) [inline vhdbuilder/packer/vhd-image-builder-sigminimal.json:298]: can be removed

File: pkg/templates/templates_generated.go [hunk 14]
```diff
@@ -5238,6 +5239,148 @@ func linuxCloudInitArtifactsSshd_config_1804_fips() (*asset, error) {
 	return a, nil
 }
 
+var _linuxCloudInitArtifactsSshd_config_2004_fips = []byte(`#	$OpenBSD: sshd_config,v 1.101 2017/03/14 07:19:07 djm Exp $
```
Comment (chengliangli0918) [inline pkg/templates/templates_generated.go:5242]: this change is not required.

**Comment by cameronmeissner** [comment]
assuming we don't need this anymore

**Pull request closed without merging**